src/types/flattened.ts

```diff
@@ -1,42 +1,43 @@
 import { MarkOptionalParams, MergeParams, RoutePathParams, RouteQueryParams } from '@/types/routeMethods'
 import { Public, Route, Routes } from '@/types/routes'
-import { Identity, UnionToIntersection } from '@/types/utilities'
+import { UnionToIntersection } from '@/types/utilities'
 
 export type Flattened<
-  TRoute extends Route | Routes,
+  TRoute extends Route,
+  TPrefix extends string = '',
+  TPathParams extends Record<string, unknown[]> = Record<never, never>,
+  TQueryParams extends Record<string, unknown[]> = Record<never, never>
+> = RouteFlat<TRoute, TPrefix, TPathParams, TQueryParams> & RouteChildrenFlat<TRoute, TPrefix, TPathParams, TQueryParams>
+
+export type FlattenedRoutes<
+  TRoute extends Routes,
   TPrefix extends string = '',
-  TPathParams extends Record<string, unknown> = Record<never, never>,
-  TQueryParams extends Record<string, unknown> = Record<never, never>
-> = Identity<
-TRoute extends Route
-  ? RouteFlat<TRoute, TPrefix, TPathParams, TQueryParams> & RouteChildrenFlat<TRoute, TPrefix, TPathParams, TQueryParams>
-  : TRoute extends Routes
-    ? UnionToIntersection<{
-      [K in keyof TRoute]: TRoute[K] extends Route
-        ? Flattened<TRoute[K], TPrefix, TPathParams, TQueryParams>
-        : Record<never, never>
-    }[number]>
+  TPathParams extends Record<string, unknown[]> = Record<never, never>,
+  TQueryParams extends Record<string, unknown[]> = Record<never, never>
+> = UnionToIntersection<{
+  [K in keyof TRoute]: TRoute[K] extends Route
+    ? Flattened<TRoute[K], TPrefix, TPathParams, TQueryParams>
     : Record<never, never>
->
+}[number]>
 
 type RouteFlat<
   TRoute extends Route,
   TPrefix extends string,
-  TPathParams extends Record<string, unknown> = Record<never, never>,
-  TQueryParams extends Record<string, unknown> = Record<never, never>
+  TPathParams extends Record<string, unknown[]> = Record<never, never>,
+  TQueryParams extends Record<string, unknown[]> = Record<never, never>
 > = TRoute extends Public<TRoute> & { name: infer Name extends string }
   ? Record<Prefix<Name, TPrefix>, MarkOptionalParams<MergeParams<RoutePathParams<TRoute, TPathParams>, RouteQueryParams<TRoute, TQueryParams>>>>
   : Record<never, never>
 
 type RouteChildrenFlat<
   TRoute extends Route,
   TPrefix extends string,
-  TPathParams extends Record<string, unknown> = Record<never, never>,
-  TQueryParams extends Record<string, unknown> = Record<never, never>
+  TPathParams extends Record<string, unknown[]> = Record<never, never>,
+  TQueryParams extends Record<string, unknown[]> = Record<never, never>
 > = TRoute extends { children: infer Children extends Routes }
   ? TRoute extends Public<TRoute> & { name: infer Name extends string }
```

Comment:
I see that this hasn't changed but I'm curious, why does the route need to be named AND public? I would expect the prefix to be updated only if named. I didn't expected the route being public to matter here?

Reply:
Um good question. I guess if it has a name that should be part of the dot notation even if its not public yeah? That's probably a bug.